### ISSUE TITLE
Run MarkDisabled function when 

### ISSUE THREAD

**Issue by masaanli** [ng-bootstrap/ng-bootstrap#1325]
### Bug description:
I don't know if this is a bug or not.
 
Now when a month in a datepicker is rendered the [markDisabled]. So when you have 3 months which you display, you have 90~ days. 90 times the markDisabled is fired and this is OK.

But now when I select a date for example (in an already rendered datepicker) I want to disable some dates again. My first thought was that I can run a function on the datepicker #template like setDisabledOnDays, but there is not a kind of function like that.

 
As a work arround iIdid the trick below! To Reinitialize the disabled days.
```
  private markDisabled() {
    this.dpDate.navigateTo({
      year: this.dpDate.months[0].year,
      month: this.dpDate.months[0].number - 3
    });

    this.dpDate.navigateTo({
      year: this.dpDate.months[0].year,
      month: this.dpDate.months[0].number + 3
    });
  }
```

Idea to create a function to run the markDisabled again?

### Link to minimally-working plunker that reproduces the issue:

You can fork a plunker from one of our [demos](https://ng-bootstrap.github.io/#/components) and use it as a starting point.
Please note that we can _not_ act on bug reports without a minimal reproduction scenario in plunker. Here is why:
https://github.com/ng-bootstrap/ng-bootstrap#you-think-youve-found-a-bug

### Version of Angular, ng-bootstrap, and Bootstrap:

Angular: 2.3.5

ng-bootstrap:
20

Bootstrap:
6

**Comment by maxokorokov** [comment]
Hey @masaanli I'm aware of the issue. But I really don't want to create a `datepicker.updateDisabledDays()`-kind of API for this. 

Trying different solutions, but would be really interested to have some details on your use case. Am I right that for you re-running `markDisabled` after selected date changes would be enough?

**Comment by masaanli** [comment]
@maxokorokov true it is enough for now, but performance wise not good offcourse :)

**Comment by timmyrosen** [comment]
I agree this is much required function. To be able to enable/disable days as you please is essential in many cases, including mine. When creating a date range for example, some days can be selected from but not to, and vice versa. Has anyone managed to create a solution for this?

**Comment by maxokorokov** [comment]
@timmyrosen, a workaround might be to provide a new instance of the `markDisabled` function when you want the changes to be taken into account. Ugly, but should work.

Here is the stackblitz: https://stackblitz.com/edit/angular-li9grd

**Comment by yidingww** [comment]
> a workaround might be to provide a new instance of the markDisabled function when you want the changes to be taken into account

@maxokorokov 

This works when u set `[markDisabled]` to the `<ngb-datepicker>`, but it does NOT work with the popup version (bind `ngbDatepicker` to an `<input>`).

I think it's better to allow call markDisabled function manually. Or, can just make `[markDisabled]` function to be called eagerly (whenever user click any cells)

**Comment by projectX21** [comment]
Nothing new here? 

I have another really ugly hack:

```
effect(() => {
  this.possibleDates();
  // ng-bootstrap caches the day cells internally, so markDisabled
  // won't be called automatically when the data changes.
  // We force the datepicker to re-render the current month by navigating to a dummy date.
  this.rerenderDatePicker();
});

private rerenderDatePicker(): void {
  // Navigating to a "dummy" date (year: 1, month: 1) forces markDisabled
  // to be evaluated again for every day.
  // Afterward, we try to go back to the last navigated month/year combination,
  // or to the current one (-1, -1 -> navigates to the current one)
  // This is an ugly hack, because we cannot call any api to re-render the days manually.
  // https://github.com/ng-bootstrap/ng-bootstrap/issues/1325
  this.datePicker()?.navigateTo({ year: 1, month: 1 });
  this.datePicker()?.navigateTo({
    year: this.lastDatePickerNavigation?.year ?? -1,
    month: this.lastDatePickerNavigation?.month ?? -1
  });
}
```
